### PR TITLE
Flink: Backport PR #10526 to v1.18 and v1.20

### PR DESCRIPTION
**Pull request by zhongqishang**
Backport PR #10526

### CODE REVIEW

**Comment by pvary** [comment]
@zhongqishang:
This could help with the backport:
```
git diff HEAD^ HEAD > /tmp/v1.19.patch
cat /tmp/v1.19.patch | sed "s/v1.19/v1.20/g" > /tmp/v1.20.patch
git apply -3 /tmp/v1.20.patch
```

File: flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/sink/TestIcebergFilesCommitter.java [hunk 1]
```diff
@@ -242,21 +246,21 @@ public void testOrderedEventsBetweenCheckpoints() throws Exception {
       RowData row1 = SimpleDataUtil.createRowData(1, "hello");
       DataFile dataFile1 = writeDataFile("data-1", ImmutableList.of(row1));
 
-      harness.processElement(of(dataFile1), ++timestamp);
+      long firstCheckpointId = 1;
+      harness.processElement(of(firstCheckpointId, dataFile1), ++timestamp);
```
Comment (zhongqishang) [inline flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/sink/TestIcebergFilesCommitter.java:250]: @pvary When apply V1.20 patch, a misalignment occurred. 

But I don't know what caused it, so I fixed it manually.

**Comment by zhongqishang** [comment]
I via the follow cmd to check the diff

```

git diff --no-index flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/sink
git diff --no-index flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/sink flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/sink

git diff --no-index flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/sink 
git diff --no-index flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/sink flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/sink
```

There is no difference between 1.19 and 1.20. 
The difference between 1.19 and 1.18 is not caused by this PR.

File: flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/sink/TestIcebergFilesCommitter.java (hunk 1 above)
Comment (pvary) [reply flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/sink/TestIcebergFilesCommitter.java:250]: I think this is related to @rodmeneses's last commit.
Could you please check @rodmeneses?
Thanks,
Peter

Comment (zhongqishang) [reply flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/sink/TestIcebergFilesCommitter.java:250]: https://github.com/apache/iceberg/commit/e6f8ab9950267c18734ba0cb2ce29867790e24fe#diff-4f79d5342dad9de47d561297e1af60aac21a498f22144ad378dc762affdbfa27L142-L143

Yes, thanks for your reminder.
It should be the modification of `TestIcebergFilesCommitter.java` that caused the inconsistent line numbers.

**Review by pvary — APPROVED** [review]

**Comment by pvary** [comment]
Merged to main. Thanks for the backport @zhongqishang!

**Pull request merged**